### ISSUE TITLE
[MRESOURCES-10] Filtering should handle java.io.File values

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resources-plugin#165]
**[Mike Perham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mperham)** opened **[MRESOURCES-10](https://issues.apache.org/jira/browse/MRESOURCES-10?redirect=false)** and commented

I'm trying to use filters along with ${basedir} to setup my unit test Derby database.  Derby must have a unique directory to build its database in and this becomes a problem in multi-module builds.

Currently I'm doing this:

```xml
<bean id="wsf.storage.dataSource" class="org.apache.commons.dbcp.BasicDataSource">
  <property name="driverClassName"><value>org.apache.derby.jdbc.EmbeddedDriver</value></property>
  <property name="url"><value>jdbc:derby:target/test/testdb/${pom.artifactId};create=true</value></property>
</bean>
```

But I'd like to do this:

```xml
<bean id="wsf.storage.dataSource" class="org.apache.commons.dbcp.BasicDataSource">
  <property name="driverClassName"><value>org.apache.derby.jdbc.EmbeddedDriver</value></property>
  <property name="url"><value>jdbc:derby:${basedir}/target/test/testdb;create=true</value></property>
</bean>
```

When I try to do the latter, I get this:

java.lang.ClassCastException: java.io.File
at org.codehaus.plexus.util.InterpolationFilterReader.read(InterpolationFilterReader.java:269)
at org.codehaus.plexus.util.InterpolationFilterReader.read(InterpolationFilterReader.java:162)
at java.io.Reader.read(Reader.java:100)
at org.codehaus.plexus.util.IOUtil.copy(IOUtil.java:212)
at org.codehaus.plexus.util.IOUtil.copy(IOUtil.java:200)
at org.apache.maven.plugin.resources.ResourcesMojo.copyFile(ResourcesMojo.java:249)
at org.apache.maven.plugin.resources.ResourcesMojo.copyResources(ResourcesMojo.java:172)
at org.apache.maven.plugin.resources.TestResourcesMojo.execute(TestResourcesMojo.java:53)
at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:415)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:531)

---

**Issue Links:**
- [MRESOURCES-19](https://issues.apache.org/jira/browse/MRESOURCES-19) classcastexception when filtering a test resource with a ${basedir} defined
 (_**"is duplicated by"**_)
- [MWAR-12](https://issues.apache.org/jira/browse/MWAR-12) Add resource filtering to war plugin
 (_**"is depended upon by"**_)

**Comment by jira-importer** [comment]
**[Jason Dillon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jason@planet57.com)** commented

Filters are useless until this has been fixed.

**Comment by jira-importer** [comment]
**[Brian Topping](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=topping)** commented

I fixed this in Plexus.  See commit #3066.  Please confirm and close.

**Comment by jira-importer** [comment]
**[Mike Perham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mperham)** commented

Brian's not a JIRA user so I can't assign the issue to him.  Can someone tell me how to query svn to display the files changed for a particular repository revision and the diff against the previous version of those files so I can review his change?

**Comment by jira-importer** [comment]
**[Jesse McConnell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jmcconnell)** commented

svn diff -r3065:3066 .

run that at the top of plexus checkout and it will tell you the changes that took it from 3065 to version 3066

**Comment by jira-importer** [comment]
**[Mike Perham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mperham)** commented

Thanks Jesse.  Looks like it got fixed by adding a toString() to ensure if the variable is not a string, it is converted into one.

**Comment by jira-importer** [comment]
**[Jesse McConnell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jmcconnell)** commented

and actually this was two commits so to see the full change (and the original class cast issue)

svn diff -r3064:3066 .

I'll dig up if this is in a plexus-utils release and change accordingly

**Comment by jira-importer** [comment]
**[Jesse McConnell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jmcconnell)** commented

fix was released in plexus-utils version 1.2 so I added a dependency to that version of the lib and this will be fixed in the next release of the plugin which will be rsn (real soon now)
